src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  onAddComment: (v: CommentData) => void;
+  isLoadingBtn: boolean;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  onAddComment,
+  isLoadingBtn,
+}) => {
+  const [formData, setFormData] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [isError, setIsError] = useState({
+    name: false,
+    email: false,
+    body: false,
+  });
+
+  const handleInputChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = e.target;
+
+    setFormData({ ...formData, [name]: value });
+    setIsError(prev => ({
+      ...prev,
+      [name]: false,
+    }));
+  };
+
+  const validate = () => {
+    const newErrors = {
+      name: !formData.name.trim(),
+      email: !formData.email.trim() || !formData.email.includes('@'),
+      body: !formData.body.trim(),
+    };
+
+    setIsError(newErrors);
+
+    return !Object.values(newErrors).some(Boolean);
+  };
+
+  const handleAddComment = (ev: React.FormEvent<HTMLFormElement>) => {
+    ev.preventDefault();
+
+    if (!validate()) {
+      return;
+    }
+
+    onAddComment(formData);
+
+    setFormData(prevFormData => ({
+      ...prevFormData,
+      body: '',
+    }));
```

Comment:
The `handleAddComment` function should reset all fields in `formData`, not just the `body`. Consider resetting `name` and `email` as well to clear the form completely after submission.